### PR TITLE
run "stack build" and fix all stack.yaml to stop beaking Haskell LSP

### PR DESCRIPTION
**Pull request by dvthai89**

### CODE REVIEW

**Review by sshine — CHANGES_REQUESTED** [review]
Thanks for your work so far.

I hope the amount of feedback given here is not overwhelming.

CI reports problems, so I'm going to address those here, too.

Here is a summary of what needs fixing:
1. I suggest you pick 0 indentation spaces.
2. I would wait with adding stack.yaml.lock for another PR.
3. CI: "[Exercises missing README.md: stack.yaml](https://travis-ci.org/github/exercism/haskell/jobs/742883479#L291)" hints that [you committed exercises/stack.yaml](https://github.com/exercism/haskell/pull/924/files#diff-a465d856718e707d84c4955514d5e5c9dbcad90bd12725d3bbb5498d279c959f) even though stack.yaml files are only supposed to go into subdirectories. This was probably an unintended by-product of an automated action.
4. CI: "[Exercises with out-of-date README.md: ...](https://travis-ci.org/github/exercism/haskell/jobs/742883479#L293)" is an Exercism-specific check you get to know about now: Since this PR does not touch or address README.md files, we don't care that they have come out of sync; this is a separate concern. We can suppress this warning by adding the message "[I have confirmed that no README check is needed](https://github.com/exercism/haskell/blob/master/bin/ensure-readmes-are-updated.sh#L3)" to any git commit message in this PR.
5. "[All exercises should have resolver lts-15.8 packages: - .](https://github.com/exercism/haskell/blob/master/.travis.yml#L42)" is a glitch in the CI script because we check that all exercises use the same resolver in a slightly crude way by assuming that stack.yaml does not contain anything but a line "`resolver: ...`". This PR breaks that assumption.

If you could address items 1-4, I will look at item 5 and reference this PR so that stack.yaml may be allowed to contain other properties than `resolver`.

**Comment by sshine** [comment]
Also, please don't close this PR and create a new one. (This is your third PR.)

Having all communication related to this issue in one thread is a good thing.

If you think the tower of commits is getting too high, you can always squash them and force push. This removes the commit history available in the PR at your discretion, but preserves messages. Feedback items can be marked as resolved so they don't take up space.

**Comment by sshine** [comment]
For record keeping: Solution to this problem is discussed here: #922.

File: exercises/accumulate/stack.yaml.lock [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+# For more information, please see the documentation at:
+#   https://docs.haskellstack.org/en/stable/lock_files
+
+packages: []
+snapshots:
+- completed:
+    size: 492015
+    url: https://raw.githubusercontent.com/commercialhaskell/stackage-snapshots/master/lts/15/8.yaml
+    sha256: 926bc3d70249dd0ba05277ff00943c0addb35b627cb641752669e7cf771310d0
+  original: lts-15.8
```
[the remark this answers is not in the document]
Comment (dvthai89) [reply exercises/accumulate/stack.yaml.lock:12]: I didn't add `stack.yaml.lock` intentionally. I'm just trying to run `stack build` to make sure everything works, it's generated automatically.

[the remark this answers is not in the document]
Comment (dvthai89) [reply exercises/accumulate/stack.yaml.lock]: I just remove all of them.

**Review by sshine — CHANGES_REQUESTED** [review]
It seems like you only changed a single file via the web API.

Since this PR was submitted, all stack.yaml files have had their `resolver: lts-...` lines updated. Sorry about complicating your workflow, but your effort motivated me to get around to fixing some things that have been broken for a year.

The easier way to continue work in this PR, if I were to do it, would be:

1. Reset the `master` branch to the latest one at "exercism/haskell":
    ```
    git remote add upstream git@github.com/exercism/haskell.git
    git fetch upstream
    git reset --hard upstream/master
    git push origin master
    ```
   (This way we're keeping three distinct git repositories in sync: "exercism/haskell" on GitHub, "dvthai89/haskell" on GitHub, and your local clone repo of "dvthai89/haskell". Your local clone repo is wired so that `origin` is a remote endpoint that points to "dvthai89/haskell". But because we're working on many things at once, you want your local clone repo to also have `upstream` as a remote endpoint that points to "exercism/haskell".)

   (**Edit:** Added `git push origin master` for good measure, since I said we keep all three in sync.)

2. Update all stack.yaml files again:
    ```sh
    cd exercises
    for e in *; do echo -e "packages:\n- .\n" >> $e/stack.yaml; done
    ```

3. Check that you got it right:
    ```sh
    git diff
    ```

    ```diff
    diff --git a/exercises/accumulate/stack.yaml b/exercises/accumulate/stack.yaml
    index 35c2e5b..b75ef3e 100644
    --- a/exercises/accumulate/stack.yaml
    +++ b/exercises/accumulate/stack.yaml
    @@ -1 +1,4 @@
     resolver: lts-16.21
    +packages:
    +- .
    +
    diff --git a/exercises/acronym/stack.yaml b/exercises/acronym/stack.yaml
    index 35c2e5b..b75ef3e 100644
    --- a/exercises/acronym/stack.yaml
    +++ b/exercises/acronym/stack.yaml
    @@ -1 +1,4 @@
     resolver: lts-16.21
    +packages:
    +- .
    +
    ...
    ```

4. Oops; realize that you added one linebreak too much, go back and fix:
    ```sh
    git checkout .
    for e in *; do echo -e "packages:\n- ." >> $e/stack.yaml; done
    git diff
    ```
   (Here `git checkout .` in the exercises/ directory will remove *all* changes that were not yet staged. And `echo -e "..."` will print `...` but with things like `\n` replaced for actual linebreaks; in shell scripting you don't normally get that, just try: `echo "\n"` -- or worse, try `echo \n` and see that that means something completely different.)

5. Don't run `git add .` (in general) because you accidentally add things you don't want to.

    ```sh
    for e in *; do git add $e/stack.yaml; done
    ```

   Or you could do the following:

    ```
    git status | grep stack.yaml | awk '{print $1}' | xargs git add
    ```

6. Commit and push the changes; because you rebased, you will need to force push:
    ```
    git commit -m "Add default 'packages' section to stack.yaml."
    git push -f
    ```

**Comment by sshine** [comment]
Hi again.

I changed my mind. ;-D

It would be much easier if we restart in another pull request.

The reason is that you are not using feature branches but you are committing from your `master` branch.

So, I'm going to close this PR and let you open another one with the following workflow:

1. Reset your current repositories (both your local and the one at "dvthai89/haskell"):
    ```
    git remote add upstream git@github.com/exercism/haskell.git
    git fetch upstream
    git reset --hard upstream/master
    git push -f origin master
    ```

2. Start a new feature branch and make your work there:
    ```
    git checkout -b add-packages-to-stack-yaml
    for e in exercises; do
      echo -e "packages:\n- ." >> exercises/$e/stack.yaml
      git add exercises/$e
    done
    git commit -m "Add default 'packages' section to stack.yaml."
    git push
    ```

3. Observe that `git` needs to know where to push this feature branch:
    ```
    fatal: The current branch add-packages-to-stack-yaml has no upstream branch.
    To push the current branch and set the remote as upstream, use

        git push --set-upstream origin add-packages-to-stack-yaml
    ```

4. Run that command instead and follow the link to make a new PR:
    ```
    git push --set-upstream origin add-packages-to-stack-yaml
    ```

    ```
    Total 0 (delta 0), reused 0 (delta 0)
    remote: 
    remote: Create a pull request for 'add-packages-to-stack-yaml' on GitHub by visiting:
    remote:      https://github.com/<you>/haskell/pull/new/add-packages-to-stack-yaml
    remote: 
    To github.com:<you>/haskell.git
     * [new branch]      add-packages-to-stack-yaml -> add-packages-to-stack-yaml
    Branch 'add-packages-to-stack-yaml' set up to track remote branch 'add-packages-to-stack-yaml' from 'origin'.
    ```

**Pull request closed without merging**